### PR TITLE
feat(verity): admin panel CRUD MVP

### PR DESCRIPTION
**Pull request by marselkhisamov**

### CODE REVIEW

**Comment by NikVolosatov** [comment]
Looks good! 🚀🚀🚀
But I suggest to move from promise callbacks to async/await. This will extremely improve the readability of the code. What do you think?

```
useEffect(() => {
  const fetchData = async () => {
    const data = await fetch();
    const json = await response.json();

    setData(json);
  }

  fetchData().catch(console.error);;
}, [])

```

File: libs/admin-panel/src/lib/components/app-component.tsx [hunk 1]
```diff
@@ -1,18 +1,68 @@
...
 
-export function AppComponent(app: AppWithVersionsAndDeps) {
+export interface AppComponentProps {
+  readonly app: AppWithVersionsAndDeps;
+  readonly updateAppList: () => void;
+  readonly onDelete: (id: string) => void;
+}
+
+export function AppComponent({ app, onDelete, updateAppList }: AppComponentProps) {
+  const [application, setItem] = useState(app);
```
Comment (marryday) [inline libs/admin-panel/src/lib/components/app-component.tsx]: Please rename setItem to setApplication

File: libs/admin-panel/src/lib/components/apps-list.tsx [hunk 2]
```diff
@@ -3,31 +3,67 @@
 import { useEffect, useState } from 'react';
 
 import { AppWithVersionsAndDeps } from '@verity/app';
-import { Button } from '@verity/ui/button';
 
 import { AppComponent } from './app-component';
+import { InputModal } from './input-modal';
 
 export default function AppsList() {
   const [apps, setItems] = useState([] as AppWithVersionsAndDeps[]);
```
Comment (marryday) [inline libs/admin-panel/src/lib/components/apps-list.tsx]: here as well

**Comment by marselkhisamov** [comment]
> Looks good! 🚀🚀🚀 But I suggest to move from promise callbacks to async/await. This will extremely improve the readability of the code. What do you think?

Nice catch, thanks!

**Comment by marryday** [comment]
> Looks good! 🚀🚀🚀 But I suggest to move from promise callbacks to async/await. This will extremely improve the readability of the code. What do you think?

useEffect(() => {
  const fetchData = async () => {
    const data = await fetch();
    const json = await response.json();

    setData(json);
  }

  fetchData().catch(console.error);;
}, [])

File: libs/admin-panel/src/lib/components/dependency-modal.tsx [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+    setIsOpen(false);
+  };
+
+  const handleVersionChange = (id: number) => {
+    const version = versions.find((version) => version.id === id) ?? null;
+    setVersion(version);
+  };
+
+  const resetState = () => {
+    if (!isOpen) {
```
Comment (marryday) [inline libs/admin-panel/src/lib/components/dependency-modal.tsx:82]: I would prefer to move the condition inside the ﻿useEffect, as it currently requires scrolling to understand that we don't actually reset the state every time ﻿isOpen changes.

Comment (marryday) [inline libs/admin-panel/src/lib/components/dependency-modal.tsx]: I would prefer to move the condition inside the ﻿useEffect, as it currently requires scrolling to understand that we don't actually reset the state every time ﻿isOpen changes.

**Review by marryday — COMMENTED** [review]
test

**Review by marryday — APPROVED** [review]

**Comment by marryday** [comment]
Well done !

**Pull request merged**